File: commandLine.lua
#!/usr/bin/env lua

local originalOptions = nil
local logFileName = 'bootstrap_installer.log'

commandLine = {}

function commandLine.parseOptions(args)
  originalOptions = args
  local continueParsing = true
  local currentArgPosition = 0
  local totalArgs = #args
  while currentArgPosition < totalArgs  do
    if args[currentArgPosition] == '--logfile' then
      logFileName = args[currentArgPosition + 1]
      currentArgPosition = currentArgPosition + 1
    end
    currentArgPosition = currentArgPosition + 1
  end
  
end

function commandLine.getLogFileName()
  return logFileName
end

return commandLine

File: commandLine.rb
#!/usr/bin/env jruby

class CommandLine
  
  COMMAND_LINE_OPTIONS = {}
  
  class << self
    
    def parseOptions(args)
      if args
        while args.size > 0
          currentArg = args.shift
          # Skip any options to do with logging
          next if currentArg =~ /logfile/ or currentArg =~ /consoleLogging/
        end
      end
    end

  end
  
end

